src/commands/force/source/deploy.ts

```diff
@@ -155,10 +155,10 @@ export class Deploy extends DeployCommand {
       this.updateDeployId(deploy.id);
 
       if (!this.isAsync) {
-        // if SFDX_USE_PROGRESS_BAR is unset or true (default true) AND we're not print JSON output
-        if (env.getBoolean('SFDX_USE_PROGRESS_BAR', true) && !this.isJsonOutput()) {
-          this.initProgressBar();
-          this.progress(deploy);
+        // we're not print JSON output
+        if (!this.isJsonOutput()) {
+          if (env.getBoolean('SFDX_USE_PROGRESS_BAR', true)) this.initProgressBar();
```

Comment:
can we move the `this.initProgressBar` call into `progress`? It would be nice to consolidate all of the relevant logic